curvefs/src/tools/curvefs_tool.h

```diff
@@ -232,9 +232,16 @@ class CurvefsToolRpc : public CurvefsTool {
             }
             // if service_stub_func_ does not assign a value
             // it will crash in there
-            service_stub_func_(controller_.get(), &requestQueue_.front(),
-                               response_.get());
-            if (controller_->Failed()) {
+            uint32_t i = 0;
+            for (i = 0; i <= FLAGS_rpcRetryTimes; ++i) {
+                controller_->Reset();
+                service_stub_func_(controller_.get(), &requestQueue_.front(),
+                                   response_.get());
+                if (!controller_->Failed()) {
+                    break;
+                }
```

Comment:
After failed you should check the retcode and check whether need retry and need sleep for a while before retry.

It needn't retry In some cases, you can refer to :https://github.com/opencurve/curve/pull/1684/files

Reply:
> After failed you should check the retcode and check whether need retry and need sleep for a while before retry.
> 
> It needn't retry In some cases, you can refer to :https://github.com/opencurve/curve/pull/1684/files

fix
This is not multi-threaded, and sleep should not be needed